From: Tony Jones <tonyj@suse.de>
Subject: remove UTRACE code from perfmon2
Patch-mainline: never, see description in perfmon2.patch

Remove UTRACE code from perfmon2.  Uses previous interface and is no longer 
present in latest perfmon2 git.   If time permits, either update to latest 
UTRACE interface or cherry pick 013f05d4bc76a2606a322261f27d7fabfd213d89
from perfmon2.git

Signed-off-by: Tony Jones <tonyj@suse.de>

---
 perfmon/perfmon_syscalls.c |   47 ---------------------------------------------
 1 file changed, 47 deletions(-)

--- a/perfmon/perfmon_syscalls.c
+++ b/perfmon/perfmon_syscalls.c
@@ -73,52 +73,6 @@
  * 	  also need to protect against timer interrupts which could trigger
  * 	  a set switch.
  */
-#ifdef CONFIG_UTRACE
-#include <linux/utrace.h>
-
-static u32
-stopper_quiesce(struct utrace_attached_engine *engine, struct task_struct *tsk)
-{
-	PFM_DBG("quiesced [%d]", tsk->pid);
-	complete(engine->data);
-	return UTRACE_ACTION_RESUME;
-}
-
-void
-pfm_resume_task(struct task_struct *t, void *data)
-{
-	PFM_DBG("utrace detach [%d]", t->pid);
-	(void) utrace_detach(t, data);
-}
-
-static const struct utrace_engine_ops utrace_ops =
-{
-	.report_quiesce = stopper_quiesce,
-};
-
-static int pfm_wait_task_stopped(struct task_struct *task, void **data)
-{
-	DECLARE_COMPLETION_ONSTACK(done);
-	struct utrace_attached_engine *eng;
-	int ret;
-
-	eng = utrace_attach(task, UTRACE_ATTACH_CREATE, &utrace_ops, &done);
-	if (IS_ERR(eng))
-		return PTR_ERR(eng);
-
-	ret = utrace_set_flags(task, eng,
-			       UTRACE_ACTION_QUIESCE | UTRACE_EVENT(QUIESCE));
-	PFM_DBG("wait quiesce [%d]", task->pid);
-	if (!ret)
-		ret = wait_for_completion_interruptible(&done);
-
-	if (ret)
-		(void) utrace_detach(task, eng);
-	else
-		*data = eng;
-	return 0;
-}
-#else /* !CONFIG_UTRACE */
 static int pfm_wait_task_stopped(struct task_struct *task, void **data)
 {
 	int ret;
@@ -139,7 +93,6 @@ static int pfm_wait_task_stopped(struct
 }
 void pfm_resume_task(struct task_struct *t, void *data)
 {}
-#endif
 
 struct pfm_syscall_cookie {
 	struct file *filp;
